src/pages/examples.tsx

```diff
@@ -166,11 +178,64 @@ const options = [
 	},
 	{
 		heading: 'Uppy',
-		options: [{ label: 'Restrictions', type: 'restrictions' }],
+		options: [
+			{ label: 'Restrictions', type: 'restrictions' },
+			{ label: 'Golden Retriever', type: 'enableGoldenRetriever' },
+		],
 	},
 ];
 
-const companionUrl = 'https://companion.uppy.io';
+const Uppy = memo(({ state, locale }) => {
+	const createUppy = useCallback(() => {
+		const ret = new UppyCore({
+			restrictions: state.restrictions,
+			locale,
+		})
+			.use(Webcam)
+			.use(ScreenCapture)
+			.use(Audio)
+			.use(ImageEditor, {})
+			.use(Tus, { endpoint })
+			.use(GoogleDrive, { companionUrl })
+			.use(Dropbox, { companionUrl })
+			.use(Instagram, { companionUrl })
+			.use(Url, { companionUrl })
+			.use(OneDrive, { companionUrl })
+			.use(Unsplash, { companionUrl })
+			.use(Box, { companionUrl });
+
+		if (state.enableGoldenRetriever) {
+			ret.use(GoldenRetriever);
+		}
+
+		return ret;
+	}, [state, locale]);
+
+	const [uppy, setUppy] = useState(createUppy);
+
+	useEffect(() => setUppy(createUppy()), [createUppy]);
```

Comment:
I think the result of using state + effect isn't any different from initliazing uppy in the render? If we use state, we should use `setOptions` in a effect, but since we're not doing that, we can keep it simple?

Reply:
I didn't want to implement setOptions in useEffect because as we've seen in previous discussions, it's quite complicated to implement it correctly. But using a useEffect and definining dependencies like this ensures that Uppy doesn't get accidentally reset if something causes a re-render (e.g. if something elsewhere in the app gets re-rendered)